Use caret-right icon instead of greater-than text symbol

jarolrod opened the pull request:
Mainly only relevant in the `AboutOptions` component for now.

| master | pr |
| ------- | -- |
| <img width="752" alt="Screen Shot 2022-08-03 at 6 45 51 AM" src="https://user-images.githubusercontent.com/23396902/182590655-4e765f99-8de4-4782-bc12-6682b63b1eeb.png"> | <img width="752" alt="Screen Shot 2022-08-06 at 7 40 39 PM" src="https://user-images.githubusercontent.com/23396902/183269335-b8dd6894-14e9-47b5-8d43-6d2f540cbb27.png"> |



[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/144)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/144)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/144)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/144)

shaavan reviewed (commented):
Concept ACK 

- I can see the visual difference between master and PR while testing.

**Screenshot:**

|Master|PR|
|---------|----|
|![Screenshot from 2022-08-03 21-04-59](https://user-images.githubusercontent.com/85434418/182650150-5bede775-0be4-458c-91e3-bb076aea6d06.png)|![Screenshot from 2022-08-03 21-05-27](https://user-images.githubusercontent.com/85434418/182650330-105020e8-4336-4482-ad84-abb3cb749be2.png)|

- This PR also solves the issue of the misaligned arrow of the developer options selector.

**Suggestion:**

- There is one more instance of setting which need to be addressed here. The ">" for the Proxy settings option on the Connection Settings page should be converted to the icon.

Rspigler commented:
tACK commit 1c17b774e6363c53ff7e1d6249ee9abb10063f7c

I can see the change and it is all lined up on my machine.  The one missing the change is the Proxy setting under 'Connection Settings' as @shaavan mentions

jarolrod commented:
updated from from 1c17b77 to 1dcdac6 ([pr144.01](https://github.com/jarolrod/gui-qml/tree/pr144.01) -> [pr144.02](https://github.com/jarolrod/gui-qml/tree/pr144.02), [diff](https://github.com/jarolrod/gui-qml/compare/pr144.01..pr144.02))

Changes: 
- rebased over changes in PR's this PR is based on
- Updated icon sources to use `image://` syntax instead of `qrc` syntax, as the `image://` syntax is required for the image provider to work and provide a proper resolution image. Comparison below (top is `image://` and below that is `qrc`:

<img width="177" alt="Screen Shot 2022-08-06 at 6 58 51 PM" src="https://user-images.githubusercontent.com/23396902/183269253-730996a9-7911-4366-9841-e6c127cd1575.png">

jarolrod commented:
updated from [1dcdac6 ](https://github.com/bitcoin-core/gui-qml/commit/1dcdac6575eaecea07a59694435541e5f106808e)to [36bc09d ](https://github.com/bitcoin-core/gui-qml/commit/36bc09d02b4f08b3e72ed3ef79c53b7cb8971e6d)[Compare](https://github.com/bitcoin-core/gui-qml/compare/1dcdac6575eaecea07a59694435541e5f106808e..36bc09d02b4f08b3e72ed3ef79c53b7cb8971e6d)

Changes: give the developer options setting button the correct action, where it will actually open the developer options subpage

jarolrod commented:
updated from [36bc09d ](https://github.com/bitcoin-core/gui-qml/commit/36bc09d02b4f08b3e72ed3ef79c53b7cb8971e6d)to [f9f6921](https://github.com/bitcoin-core/gui-qml/commit/f9f692136f441ecee2686424c7fc38593bf44c82)

changes: updated over changes on master

hebasto commented on src/qml/components/AboutOptions.qml:
```diff
@@ -13,42 +13,56 @@ ColumnLayout {
-            bold: false
-            rightalign: true
+        actionItem: Button {
+            icon.source: "image://images/caret-right"
+            icon.color: Theme.color.neutral9
+            icon.height: 18
+            icon.width: 18
+            background: null
```
Why are the `icon.color` and `background` properties are specified for some elements, but not for others?

shaavan replied:
Button is a qml type, so we need to specify the `icon.color`, background value explicitly for each time the Button is used as `actionItem`, which this PR does.

Whereas, the `ExternalLink` is a custom-defined component that already employs the respective icon.color, and background value in its definition, so we need not specify that separately when using it.

In `src/qml/controls/ExternalLink.qml`

```qml
            sourceComponent: Button {
                icon.source: root.iconSource
                icon.color: Theme.color.neutral9
                icon.height: root.iconHeight
                icon.width: root.iconWidth
                background: null
                onClicked: Qt.openUrlExternally(link)
            }
```

shaavan reviewed (commented):
ACK f9f692136f441ecee2686424c7fc38593bf44c82

The `ExternalLink` and `Button` components replace all the instances of _greater-than text symbol_ and are used appropriately.

I have two ideas to discuss that we can consider incorporating in this PR or in a follow-up.

1. The `Button{}` is used twice as an `actionItem`  in the current implementation of GUI. There is a chance that we could need to use this Button configuration multiple times as the development progresses. So we can think about extracting this code as a separate component.
2. There is only one instance of usage of `ExternalLink{}` where the default values for `iconWidth`, and `iconHeight` `(30, 30)` are used. All other usage instances modify these values to `(18, 18)`. Changing the default values in `ExternalLink.qml` for these properties will save up on redundant code.
